crates/core/tedge_actors/src/message_boxes.rs

```diff
@@ -142,44 +143,53 @@ pub trait MessageBox: 'static + Sized + Send + Sync {
     fn logging_is_on(&self) -> bool;
 }
 
+/// Either a message or a [RuntimeRequest]
+pub enum WrappedInput<Input> {
+    Message(Input),
+    RuntimeRequest(RuntimeRequest),
+}
+
+#[async_trait]
+pub trait ReceiveMessages<Input: Debug> {
+    /// Return the next received message if any, returning [RuntimeRequest]'s as errors.
+    /// Returning [RuntimeRequest] takes priority over messages.
+    async fn try_recv(&mut self) -> Result<Option<Input>, RuntimeRequest>;
+
+    /// Returns [Some] [WrappedInput] the next time a message is received. Returns [None] if
+    /// the underlying channels are closed. Returning [RuntimeRequest] takes priority over messages.
+    async fn recv_message(&mut self) -> Option<WrappedInput<Input>>;
+
+    /// Returns [Some] message the next time a message is received. Returns [None] if
+    /// either of the underlying channels are closed.
+    /// Returning [RuntimeRequest] takes priority over messages.
+    async fn recv(&mut self) -> Option<Input>;
+}
+
 /// The basic message box
 pub struct SimpleMessageBox<Input, Output> {
     name: String,
-    input_receiver: mpsc::Receiver<Input>,
-    signal_receiver: mpsc::Receiver<RuntimeRequest>,
+    receiver: CombinedReceiver<Input>,
     output_sender: DynSender<Output>,
     logging_is_on: bool,
 }
 
 impl<Input: Message, Output: Message> SimpleMessageBox<Input, Output> {
     pub fn new(
         name: String,
-        input_receiver: mpsc::Receiver<Input>,
-        signal_receiver: mpsc::Receiver<RuntimeRequest>,
+        receiver: CombinedReceiver<Input>,
         output_sender: DynSender<Output>,
     ) -> Self {
         SimpleMessageBox {
             name,
-            input_receiver,
-            signal_receiver,
+            receiver,
             output_sender,
             logging_is_on: true,
         }
     }
 
+    // TODO: Should be removed
     pub async fn recv(&mut self) -> Option<Input> {
```

Comment:
Why not remove it now as we have `impl<I,O> ReceiveMessages<I> for SimpleMessageBox<I,O>`